app/main.py

```diff
@@ -1,2 +1,72 @@
+from typing import Any
+
+INC_SIZE_TWICE = 2
+
+
 class Dictionary:
-    pass
+    size = 8
+
+    def __init__(self) -> None:
+        self.capacity = 0
+        self.hash_table = [None] * Dictionary.size
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if isinstance(key, (list, dict, set)):
+            raise KeyError("key must be hashable")
+        hash_key = hash(key)
+
+        # rewrite value, if we have the same key
+        for item in self.hash_table:
+            if item and item.hash == hash_key and item.key == key:
+                item.value = value
+                return
```

Comment:
Code Efficiency: This loop might have O(n) time complexity in the worst case. Consider using a more efficient data structure or algorithm to improve the performance.